lib/livebook/file_system/s3/client.ex

```diff
@@ -263,6 +263,11 @@ defmodule Livebook.FileSystem.S3.Client do
     %{host: host} = URI.parse(file_system.bucket_url)
     headers = [{"Host", host} | headers]
 
+    headers =
+      if file_system.session_token,
+        do: [{"X-Amz-Security-Token", file_system.session_token} | headers],
+        else: headers
+
```

Comment:
From an Elixir style perspective, is it better to do this kind of double assignment, or to do something like:

```elixir
    headers = [{"Host", host} | headers] |> add_security_token(file_system)

    ...

  defp add_security_token(headers, %{session_token: nil}), do: headers
  defp add_security_token(headers, %{session_token: token}), do: [{"X-Amz-Security-Token", file_system.session_token} | headers]
```

Reply:
Both are fine, in this example I am totally happy with having it inlined instead of a separate function.